File: src/diff_bot_py/diff_bot_py/simple_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimplePublisher(Node):
    def __init__(self):
        super().__init__('simple_publisher')
        self.publisher_ = self.create_publisher(String, 'topic', 10)

        self.counter_ = 0
        self.frequency_ = 1.0  # Frequency in Hz
        self.get_logger().info('Publishing at %d Hz' % self.frequency_)
        self.timer = self.create_timer(1.0 / self.frequency_, self.timer_callback)

    def timer_callback(self):
        msg = String()
        msg.data = 'Hello, ROS 2! count:%d' % self.counter_
        self.publisher_.publish(msg)
        self.counter_ += 1
        self.get_logger().info(f'Publishing: "{msg.data}"')



def main():
    rclpy.init()
    simple_publisher = SimplePublisher()
    rclpy.spin(simple_publisher)
    simple_publisher.destroy_node()
    rclpy.shutdown()  

if __name__ == '__main__':
    main()          

File: src/diff_bot_cpp/src/simple_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>
#include <chrono>


using namespace std::chrono_literals;

class SimplePublisher : public rclcpp::Node
{
  public:

     SimplePublisher(): Node("simple_publisher")
     {
       publisher_ = this->create_publisher<std_msgs::msg::String>("topic", 10);
       timer_ = this->create_wall_timer(
         std::chrono::seconds(1),
         std::bind(&SimplePublisher::timerCallback, this)
        
        );
    }
      
     private:
        unsigned int counter_;
        
        rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
         
        void timerCallback()
        {
        auto message = std_msgs::msg::String();
        message.data = "Hello, ROS2!";
        RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
        publisher_->publish(message);
        }
};

int main (int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<SimplePublisher>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}